blocks/carousel/carousel.js

```diff
@@ -0,0 +1,143 @@
+import { fetchPlaceholders } from '../../scripts/aem.js';
+
+let tId;
+function debounce(method, delay) {
+  clearTimeout(tId);
+  tId = setTimeout(() => {
+    method();
+  }, delay);
+}
+
+function showSlide(block, slideIndex = 0, scroll = true) {
+  const slides = block.querySelectorAll('.carousel-slide');
+  slides.forEach((slide) => slide.setAttribute('aria-hidden', 'true'));
+
+  let realSlideIndex = slideIndex < 0 ? slides.length - 1 : slideIndex;
+  if (slideIndex >= slides.length) realSlideIndex = 0;
+  const activeSlide = slides[realSlideIndex];
+
+  activeSlide.setAttribute('aria-hidden', 'false');
+  if (scroll) {
+    block.querySelector('.carousel-slides').scrollTo({
+      top: 0,
+      left: activeSlide.offsetLeft,
+      behavior: 'smooth',
+    });
+  }
+
+  const indicators = block.querySelectorAll('.carousel-slide-indicator');
+  indicators.forEach((indicator) => {
+    indicator.querySelector('button').removeAttribute('disabled');
+  });
+  const activeIndicator = indicators[realSlideIndex];
+  activeIndicator.querySelector('button').setAttribute('disabled', 'true');
+
+  block.dataset.activeSlide = realSlideIndex;
+}
+
+function bindEvents(block) {
+  const slideIndicators = block.querySelector('.carousel-slide-indicators');
+  slideIndicators.querySelectorAll('button').forEach((button) => {
+    button.addEventListener('click', (e) => {
+      const slideIndicator = e.currentTarget.parentElement;
+      showSlide(block, parseInt(slideIndicator.dataset.targetSlide, 10));
+    });
+  });
+
+  block.querySelector('.slide-prev').addEventListener('click', () => {
+    showSlide(block, parseInt(block.dataset.activeSlide, 10) - 1);
+  });
+  block.querySelector('.slide-next').addEventListener('click', () => {
+    showSlide(block, parseInt(block.dataset.activeSlide, 10) + 1);
+  });
+
+  const slidesWrapper = block.querySelector('.carousel-slides');
+  slidesWrapper.addEventListener('scroll', () => {
+    debounce(() => {
+      const position = slidesWrapper.scrollLeft;
+      const slideWidth = slidesWrapper.querySelector('.carousel-slide').scrollWidth;
+      const slide = Math.round(position / slideWidth);
+      showSlide(block, slide, false);
+    }, 200);
+  });
+}
+
+function createSlide(row, slideIndex, carouselId, placeholders) {
+  const slide = document.createElement('li');
+  slide.setAttribute('role', 'group');
```

Comment:
I'm confused by this.

Reply:
Or because it's in a semantic list, do we need this at all?

Reply:
I pulled this from w3c guideline on carousel, I think the idea is to convey that the elements inside the li are a group of items representing a slide in the carousel, but you are right?

https://www.w3.org/WAI/ARIA/apg/patterns/carousel/#wai-ariaroles,states,andproperties

`Each slide container has role [group](https://w3c.github.io/aria/#group) with the property [aria-roledescription](https://w3c.github.io/aria/#aria-roledescription) set to slide`